includes/Admin/Settings_Screens/Product_Sync.php

```diff
@@ -174,14 +174,17 @@ class="button button-large button-primary"
 	 * @param array $field field data
 	 */
 	public function render_title( $field ) {
+		$integration = facebook_for_woocommerce()->get_integration();
 
+		$show_sync_button = $integration->allow_full_batch_api_sync() &&
+			facebook_for_woocommerce()->get_connection_handler()->is_connected();
 		?>
 
 		<h2>
 
 			<?php esc_html_e( 'Product sync', 'facebook-for-woocommerce' ); ?>
 
-			<?php if ( facebook_for_woocommerce()->get_connection_handler()->is_connected() ) : ?>
+			<?php if ( $show_sync_button ) : ?>
```

Comment:
As discussed in #2037 - I think it's clearer and simpler to leave the button and show the error message.